src/db/migrations/3-create-device.sql

```diff
@@ -0,0 +1,6 @@
+CREATE TABLE devices (
+    device_id INTEGER PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+    user_id INTEGER NOT NULL,
+    device_token TEXT NOT NULL,
```

Comment:
This pair - `(user_id, device_token)` - is the [natural key](https://en.wikipedia.org/wiki/Natural_key) for this table. At the very least, we should have a unique constraint on the table.

I've spent a few minutes going back and forth about whether or not we need the [surrogate key](https://en.wikipedia.org/wiki/Surrogate_key) (`device_id`), or if we should make `(user_id, device_token)` the primary key. I don't think we'll ever need to update device tokens; as I understand the Firebase documentation, they're effectively immutable. I don't think the client (neither the mobile app nor the PHP API) should know or care about the `device_id` we've assigned. I believe the only way we'll ever delete a token is when we try to use it and Firebase tells us the token is no longer valid, at which point we can simply `DELETE FROM devices WHERE device_token = ?`. The main argument in favor is "but what if...?", but I can't think of any reasons we'd need to uniquely address a token when we don't already have the token; eg I don't think we'll need `GET/PUT/DELETE /api/devices/123`. Can you?

So yeah, writing this out, I think we should drop `device_id` and make our primary key `(user_id, device_token)`.